### PR TITLE
Navbar revamp

### PR DESCRIPTION
**Pull request by yangshun**

### CODE REVIEW

**Comment by tgxworld** [comment]
@yangshun Thanks :+1: This part looks good to me. 

Just to check, do we need to include the bootstrap's JS files for the "hamburger" on mobile screen sizes to work?

**Comment by yangshun** [comment]
Ah yes. Oops I forgot to add that in >_<. I'll fix that after I get the damned jquery to be found

**Comment by yangshun** [comment]
Ok added `bootstrap/transition` and `bootstrap/collapse` (:

**Comment by tgxworld** [comment]
Cool looks good to me :) I'll merge it first?

File: app/views/layouts/_top_nav.html.haml [hunk 1]
```diff
@@ -1,18 +1,24 @@
...
-    %li= link_to t('top_nav.benchmarks'),
-      repo_path(organization_name: 'ruby', repo_name: 'ruby')
-
-    %li= link_to t('top_nav.hardware'), hardware_path
-    %li= link_to t('top_nav.contribute'), contribute_path
-    %li= link_to t('top_nav.sponsors'), sponsors_path
-
-    %li
-      = link_to t('top_nav.discuss'), AppSettings.forum_url, target: '_blank'
+    .collapse.navbar-collapse{:id => 'rbn-navbar'}
```
Comment (tgxworld) [inline app/views/layouts/_top_nav.html.haml]: Lets use `id: 'rbn-navbar'` here. `:id => 'rbn-navbar` is using Ruby's 1.9 syntax which is usually not preferred.

File: app/views/layouts/_top_nav.html.haml [hunk 2]
```diff
@@ -1,18 +1,24 @@
-.row
-  %h1.top-nav-title.l-left
-    = link_to root_path do
-      = t('top_nav.logo')
+%nav.navbar.navbar-default
+  .container
+    .navbar-header
+      %button.navbar-toggle.collapsed{:data => {:toggle => 'collapse', :target => '#rbn-navbar'}}
```
Comment (tgxworld) [inline app/views/layouts/_top_nav.html.haml]: Ditto: https://github.com/ruby-bench/ruby-bench-web/pull/66/files#r30198478

File: app/views/layouts/_top_nav.html.haml (hunk 1 above)
Comment (yangshun) [inline app/views/layouts/_top_nav.html.haml]: I did try that but HAML doesn't support that..

Comment (yangshun) [inline app/views/layouts/_top_nav.html.haml]: Are you sure that's achievable in HAML?

Comment (tgxworld) [inline app/views/layouts/_top_nav.html.haml]: Opps XD. haha you can do this with for id `#rbn-navbar`

Comment (yangshun) [inline app/views/layouts/_top_nav.html.haml]: Yes I also tried that. But I feel it gets lost among the `.collapse.navbar-collapse#rbn-navbar` if they are concatenated into one string. I really like the old (?) syntax as it's closer to the generated HTML where attributes and values are clearly separated.

Comment (tgxworld) [inline app/views/layouts/_top_nav.html.haml]: Ic. To me `#rbn-navbar` is just shorter to type but it isn't too big of a concern for me. I merge hor?

Comment (yangshun) [inline app/views/layouts/_top_nav.html.haml]: OK thanks. :smile:  Working on the other sections now!

Comment (tgxworld) [inline app/views/layouts/_top_nav.html.haml]: Thanks :) Treat you lunch!!

**Pull request merged**